plugins/outputs/azure_data_explorer/azure_data_explorer.go

```diff
@@ -219,6 +312,12 @@ func (adx *AzureDataExplorer) Init() error {
 		return errors.New("Metrics grouping type is not valid")
 	}
 
+	if adx.IngestionType == "" {
+		adx.IngestionType = queuedIngestion
+	} else if !(choice.Contains(adx.IngestionType, []string{managedIngestion, queuedIngestion})) {
+		return fmt.Errorf("unknown ingestion type %q", adx.IngestionType)
+	}
```

Comment:
Can this check please be performed in the `Init()` function to do it only once?

Reply:
We looked at some of the other output plugins eg [Cloudwatch_logs](https://github.com/influxdata/telegraf/blob/master/plugins/outputs/cloudwatch_logs/cloudwatch_logs.go#L89), and followed the same approach, which suggests that `Init()` should be the place where all the plugin initialization happens. Are we missing something here?

Reply:
I know we do have some plugins pre-dating the existence of `Init()`, but we should use the possibilities offered by new infrastructure when adding code. So please move checks of config options to `Init()` and fail if something is wrong so that Telegraf can fail early instead of printing errors over and over!

Reply:
As I am understanding the asks here, you are suggesting moving the check to Init(), but this is already in the Init() function.  Please correct me if I am misunderstanding.

Reply:
Sorry for the noise, now I see that this _is_ the `Init()`...